File: wheel.java
public class Wheel {

    private int capacity;
    private int[] datas;
    private int end;

    Wheel(int capacity) {
        this.capacity = capacity;
        this.datas = new int[capacity];
        this.end = 0;
    }

    public void addData(int data) {
        if (isFull()) {
            System.out.println("Wheel is full");
        } else {
            this.datas[this.end] = data;
            this.end += 1;            
        }
    }

    public int getData() {
        int middle = (this.capacity - 1) / 2;
        return this.datas[middle];
    }

    public boolean isFull() {
        return (this.end == this.capacity - 1);
    }

    public boolean isEmpty() {
        return (this.end == 0);
    }

    public void rotateClockWise() {
        this.capacity -= 2;
    }

    public void rotateCounterClockWise() {
        this.capacity += 2;
    }

}

public static void main(String[] args) {
    Wheel roda = new Wheel(10);
    roda.addData(5);
    roda.addData(12);
    roda.addData(70);
    roda.addData(54);
    int x = roda.getData(); // return null karena data yang ditengah wheel masih kosong
    roda.addData(72);
    int a = roda.getData(); // return 72
    roda.rotateClockWise();
    int y = roda.getData(); // return 54
    roda.rotateClockWise();
    int b = roda.getData(); // return 70
}

File: wheel.py
class Wheel:
    """Wheel Abstract Data Type"""
    def __init__(self, capacity):
        self.capacity = capacity
        self.datas = [None for x in range(0, capacity)]
        self.end = 0

    def addData(self, data):
        # add data to end of list
        if self.isFull():
            print "Wheel is full"
        else:
            self.datas[self.end] = data
            self.end += 1

    def getData(self):
        # get data in the middle of circle
        middle = (self.capacity - 1) / 2
        return self.datas[middle]

    def isFull(self):
        # are the circle is full?
        return self.end == self.capacity - 1

    def isEmpty(self):
        # are the circle is empty?
        return self.end == 0

    def rotateClockWise(self):
        # rotating clockwise
        self.capacity -= 2

    def rotateCounterClockWise(self):
        # rotating counter-clockwise
        self.capacity += 2
